http/basic/src/main/java/org/wildfly/security/http/basic/BasicAuthenticationMechanism.java

```diff
@@ -134,6 +137,10 @@ public void evaluateRequest(final HttpServerRequest request) throws HttpAuthenti
         }
 
         List<String> authorizationValues = request.getRequestHeaderValues(AUTHORIZATION);
+        if (silent && authorizationValues == null) {
```

Comment:
another minor, if you can see below, there is this line:
`return;     //CHALLENGE NOT SENT`

this is actually the place where it is decided that the challenge does not have to be send. So would be better to add it there.

Reply:
Your code does work as well, it is just better to keep it at one place for readability

Reply:
Oh I just saw this comment, let me update this. Thanks.

Reply:
Fixed this one and pushed my changes